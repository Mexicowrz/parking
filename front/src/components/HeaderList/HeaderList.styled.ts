import styled from 'styled-components';

const Header = styled.div`
  display: flex;
  padding: 10px 0;
`;

export { Header };
