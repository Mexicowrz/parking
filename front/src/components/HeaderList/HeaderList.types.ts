/**
 * Тип данных, принимаемых компонентом заголовка списков
 */
export type HeaderListProps = {
  onSearch?: (query: string) => void;
  onAdd?: () => void;
  onEdit?: () => void;
  onDelete?: () => void;
  isDisabled?: boolean;
}