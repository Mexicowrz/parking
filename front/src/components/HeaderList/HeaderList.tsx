import React from 'react';
import { Input, Button, Popconfirm } from 'antd';
import { PlusOutlined, DeleteOutlined, EditOutlined } from '@ant-design/icons';
import * as Styled from './HeaderList.styled';
import { HeaderListProps } from './HeaderList.types';

/**
 * Компонент отображения заголовка списков
 * @param {HeaderListProps} props
 * @returns
 */
export const HeaderList: React.FC<HeaderListProps> = (
  props: HeaderListProps,
) => {
  return (
    <Styled.Header>
      <Input.Search
        data-testid='search-input'
        style={{ width: 200, marginRight: 10 }}
        onSearch={props.onSearch}
      />
      <Button
        data-testid='add-button'
        type='primary'
        shape='circle'
        icon={<PlusOutlined />}
        onClick={props.onAdd}
        style={{ marginRight: 10 }}
      />
      <Button
        data-testid='edit-button'
        type='primary'
        disabled={props.isDisabled}
        onClick={props.onEdit}
        shape='circle'
        icon={<EditOutlined />}
        style={{ marginRight: 10 }}
      />
      <Popconfirm
        title='Вы уверены, что хотите удалить пользователя?'
        onConfirm={props.onDelete}
        okText='Да'
        cancelText='Нет'
        disabled={props.isDisabled}
      >
        <Button
          data-testid='delete-button'
          type='primary'
          disabled={props.isDisabled}
          shape='circle'
          icon={<DeleteOutlined />}
        />
      </Popconfirm>
    </Styled.Header>
  );
};
