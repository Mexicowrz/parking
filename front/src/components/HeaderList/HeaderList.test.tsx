import { render, cleanup, fireEvent, within } from '@testing-library/react';
import { HeaderList } from './HeaderList';

afterEach(cleanup);

describe('Headerlist component', () => {
  it('with default props', () => {
    const { asFragment } = render(<HeaderList />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('with disabled state', () => {
    const { asFragment } = render(<HeaderList isDisabled={true} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('check add button click', () => {
    const onAdd = jest.fn();
    const { getByTestId } = render(<HeaderList onAdd={onAdd} />);
    fireEvent.click(getByTestId('add-button'));
    expect(onAdd).toBeCalled();
  });
  it('check edit button click', () => {
    const onEdit = jest.fn();
    const { getByTestId } = render(<HeaderList onEdit={onEdit} />);
    fireEvent.click(getByTestId('edit-button'));
    expect(onEdit).toBeCalled();
  });
  it('check delete button click', () => {
    const onDelete = jest.fn();
    const { getByTestId } = render(<HeaderList onDelete={onDelete} />);
    const { getByText: getByBodyText } = within(document.body);
    fireEvent.click(getByTestId('delete-button'));
    const yesButton = getByBodyText('Да');
    expect(yesButton).toBeInTheDocument();
    fireEvent.click(yesButton);
    expect(onDelete).toBeCalled();
  });
  it('check search by text', () => {
    const onSearch = jest.fn();
    const { getByTestId } = render(<HeaderList onSearch={onSearch} />);
    const input = getByTestId('search-input');
    fireEvent.change(input, { target: { value: 'test' } });
    fireEvent.keyDown(input, { key: 'Enter', code: 'Enter', keyCode: 13 });
    expect(onSearch).toBeCalled();
    expect(onSearch.mock.calls[0][0]).toBe('test');
  });
});
