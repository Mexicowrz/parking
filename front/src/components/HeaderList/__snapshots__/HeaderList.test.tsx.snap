// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headerlist component with default props 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ jcZDfb"
  >
    <span
      class="ant-input-group-wrapper ant-input-search"
      style="width: 200px; margin-right: 10px;"
    >
      <span
        class="ant-input-wrapper ant-input-group"
      >
        <input
          class="ant-input"
          data-testid="search-input"
          type="text"
          value=""
        />
        <span
          class="ant-input-group-addon"
        >
          <button
            class="ant-btn ant-btn-icon-only ant-input-search-button"
            type="button"
          >
            <span
              aria-label="search"
              class="anticon anticon-search"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="search"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                />
              </svg>
            </span>
          </button>
        </span>
      </span>
    </span>
    <button
      class="ant-btn ant-btn-primary ant-btn-circle ant-btn-icon-only"
      data-testid="add-button"
      style="margin-right: 10px;"
      type="button"
    >
      <span
        aria-label="plus"
        class="anticon anticon-plus"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="plus"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <defs>
            <style />
          </defs>
          <path
            d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
          />
          <path
            d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
          />
        </svg>
      </span>
    </button>
    <button
      class="ant-btn ant-btn-primary ant-btn-circle ant-btn-icon-only"
      data-testid="edit-button"
      style="margin-right: 10px;"
      type="button"
    >
      <span
        aria-label="edit"
        class="anticon anticon-edit"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="edit"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
          />
        </svg>
      </span>
    </button>
    <button
      class="ant-btn ant-btn-primary ant-btn-circle ant-btn-icon-only"
      data-testid="delete-button"
      type="button"
    >
      <span
        aria-label="delete"
        class="anticon anticon-delete"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="delete"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
          />
        </svg>
      </span>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Headerlist component with disabled state 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ jcZDfb"
  >
    <span
      class="ant-input-group-wrapper ant-input-search"
      style="width: 200px; margin-right: 10px;"
    >
      <span
        class="ant-input-wrapper ant-input-group"
      >
        <input
          class="ant-input"
          data-testid="search-input"
          type="text"
          value=""
        />
        <span
          class="ant-input-group-addon"
        >
          <button
            class="ant-btn ant-btn-icon-only ant-input-search-button"
            type="button"
          >
            <span
              aria-label="search"
              class="anticon anticon-search"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="search"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                />
              </svg>
            </span>
          </button>
        </span>
      </span>
    </span>
    <button
      class="ant-btn ant-btn-primary ant-btn-circle ant-btn-icon-only"
      data-testid="add-button"
      style="margin-right: 10px;"
      type="button"
    >
      <span
        aria-label="plus"
        class="anticon anticon-plus"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="plus"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <defs>
            <style />
          </defs>
          <path
            d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
          />
          <path
            d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
          />
        </svg>
      </span>
    </button>
    <button
      class="ant-btn ant-btn-primary ant-btn-circle ant-btn-icon-only"
      data-testid="edit-button"
      disabled=""
      style="margin-right: 10px;"
      type="button"
    >
      <span
        aria-label="edit"
        class="anticon anticon-edit"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="edit"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
          />
        </svg>
      </span>
    </button>
    <span
      class="ant-popover-disabled-compatible-wrapper"
      style="display: inline-block; cursor: not-allowed;"
    >
      <button
        class="ant-btn ant-btn-primary ant-btn-circle ant-btn-icon-only"
        data-testid="delete-button"
        disabled=""
        style="pointer-events: none;"
        type="button"
      >
        <span
          aria-label="delete"
          class="anticon anticon-delete"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="delete"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
            />
          </svg>
        </span>
      </button>
    </span>
  </div>
</DocumentFragment>
`;
